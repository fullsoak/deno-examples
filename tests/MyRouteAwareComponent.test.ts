import { ssr } from "fullsoak";
import { assertSnapshot } from "@std/testing/snapshot";
import { MyRouteAwareComponent } from "../src/components/MyRouteAwareComponent/index.tsx";

Deno.test("MyRouteAwareComponent", async (t) => {
  const output = await ssr(MyRouteAwareComponent, {
    url: "http://localhost/bar",
  });
  await assertSnapshot(t, output);
});
