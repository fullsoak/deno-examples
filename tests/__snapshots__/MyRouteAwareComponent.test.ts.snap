export const snapshot = {};

snapshot[`MyRouteAwareComponent 1`] = `
'<!doctype html><html><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0"><title></title><script type="importmap">
{
  "imports": {
    "preact": "https://esm.sh/preact@10.25.4",
    "preact/hooks": "https://esm.sh/preact@10.25.4/hooks",
    "react": "https://esm.sh/preact@10.25.4",
    "react/jsx-runtime": "https://esm.sh/preact@10.25.4/jsx-runtime",
    "preact/jsx-runtime": "https://esm.sh/preact@10.25.4/jsx-runtime",
    "react-dom": "https://esm.sh/preact@10.25.4/compat/",
    "react-dom/*": "https://esm.sh/preact@10.25.4/compat/*",
    "htm/preact": "https://esm.sh/htm@3.1.1/preact?external=preact",
    "preact-iso": "https://esm.sh/preact-iso?external=preact",
    "fullsoak/preact-iso": "https://esm.sh/preact-iso?external=preact",
    "@fullsoak/fullsoak/preact-iso": "https://esm.sh/preact-iso?external=preact",
    "fullsoak": "/fullsoak",
    "@fullsoak/fullsoak": "/fullsoak"
  }
}
</script><script type="text/javascript"></script><style></style></head><body><main id="fullsoak-htmlshell-main">,,[object Promise]</main><script type="text/javascript">window.preloadedProps = {"url":"http://localhost/bar"}</script><script type="module" src="/js/MyRouteAwareComponent/mount.js"></script></body></html>'
`;
